File: python/src/tools/feature_flags/get_all.py
import json
from pydantic import BaseModel
from src.tools.types import Context, Tool, ToolResult, TextContent


class GetAllFeatureFlagsParams(BaseModel):
    pass


async def get_all_feature_flags_handler(context: Context, _params: GetAllFeatureFlagsParams) -> ToolResult:
    project_id = await context.get_project_id()
    flags_result = await context.api.feature_flags(project_id).list()

    if not flags_result.success:
        raise Exception(f"Failed to get feature flags: {flags_result.error}")

    return ToolResult(content=[TextContent(text=json.dumps([flag.model_dump() for flag in flags_result.data]))])


def get_all_feature_flags_tool() -> Tool[GetAllFeatureFlagsParams]:
    return Tool(
        name="feature-flags-get-all",
        description="Use this tool to get all feature flags for the project.",
        schema=GetAllFeatureFlagsParams,
        handler=get_all_feature_flags_handler
    )

File: typescript/src/tools/surveys/getAll.ts
import { SurveyGetAllSchema } from "@/schema/tool-inputs";
import { formatSurveys } from "@/tools/surveys/utils/survey-utils";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = SurveyGetAllSchema;
type Params = z.infer<typeof schema>;

export const getAllHandler = async (context: Context, params: Params) => {
	const projectId = await context.stateManager.getProjectId();

	const surveysResult = await context.api.surveys({ projectId }).list(params ? { params } : {});

	if (!surveysResult.success) {
		throw new Error(`Failed to get surveys: ${surveysResult.error.message}`);
	}

	const formattedSurveys = formatSurveys(surveysResult.data, context, projectId);

	const response = {
		results: formattedSurveys,
	};

	return {
		content: [{ type: "text", text: JSON.stringify(response) }],
	};
};

const definition = getToolDefinition("surveys-get-all");

const tool = (): Tool<typeof schema> => ({
	name: "surveys-get-all",
	description: definition.description,
	schema,
	handler: getAllHandler,
	annotations: {
		destructiveHint: false,
		idempotentHint: true,
		openWorldHint: true,
		readOnlyHint: true,
	},
});

export default tool;
